easybuild/tools/include.py

```diff
@@ -170,12 +170,23 @@ def include_easyblocks(tmpdir, paths):
     _log.debug("Included generic easyblocks: %s", included_generic_ebs)
     _log.debug("Included software-specific easyblocks: %s", included_ebs)
 
-    # inject path into Python search path, and reload modules to get it 'registered' in sys.modules
+    # prepend new location to Python search path
     sys.path.insert(0, easyblocks_path)
-    reload(easybuild)
-    if 'easybuild.easyblocks' in sys.modules:
-        reload(easybuild.easyblocks)
-        reload(easybuild.easyblocks.generic)
+    fixup_namespace_packages(easyblocks_path)
+
+    # make sure easybuild.easyblocks(.generic) are imported;
+    # reload them to pick up all possible paths they may be in
+    import easybuild.easyblocks
+    reload(easybuild.easyblocks)
+    import easybuild.easyblocks.generic
+    reload(easybuild.easyblocks.generic)
+
+    # hard inject location to included (generic) easyblocks into Python search path
+    # only prepending to sys.path is not enough due to 'declare_namespace' in easybuild/easyblocks/__init__.py
+    new_path = os.path.join(easyblocks_path, 'easybuild', 'easyblocks')
+    easybuild.easyblocks.__path__.insert(0, new_path)
+    new_path = os.path.join(new_path, 'generic')
+    easybuild.easyblocks.generic.__path__.insert(0, new_path)
```

Comment:
In those cases I had undefined `EASYBUILD_INCLUDE_EASYBLOCKS` because I thought there was something specifically related to the easyblocks in the error